packages/sui-decorators/src/decorators/cache/index.js

```diff
@@ -20,13 +20,13 @@ const _cache = ({
   target,
   ttl
 } = {}) => {
-  caches[fnName] =
-    caches[fnName] ||
+  caches[`${target.constructor.name}::${fnName}`] =
```

Comment:
As the cacheKey now is a bit more complex, we might want to create it once.

```
const cacheKey = `${target.constructor.name}::${fnName}`
const cacheEntry = caches[cacheKey] || 
  (algorithm === ALGORITHMS.LRU
    ? new LRU({size})
    : new Error(`[sui-decorators::cache] unknown algorithm: ${algorithm}`))

caches[cacheKey] = cacheEntry
```

Reply:
I guess is not relevant in this case to build a `cacheKey` that Babel will transpile the final name of `target.constructor.name`